src/error.rs

```diff
@@ -18,15 +18,15 @@ pub enum Error {
     /// Error in underlying RustTLS library.
     /// Might signal problem with establishing secure connection using trusted certificates
     #[error("TLS error")]
-    TLSError(rustls::Error),
+    TLSError(#[source] rustls::Error),
```

Comment:
Should we also display the source in the string representation for these variants?

Reply:
I would tend to not doing this, which is also supported by https://blog.rust-lang.org/inside-rust/2021/07/01/What-the-error-handling-project-group-is-working-towards.html#guidelines-for-implementing-displayfmt-and-errorsource which suggests you should never do that at the same time as providing a source.

Reply:
Hmm, ok.